### ISSUE TITLE
Standardize platform API handling

### ISSUE THREAD

**Issue by gwhigs** [gwhigs/pbridge#1]
There should be a generic 'platform api' class, and then a platform-specific class that inherits from it for each platform.

This allows us to avoid having an "if platform==...elif platform==..." for every platform when all we need is a generic response from whatever api is attached to the current object.

Example: calling a function that grabs as much initial show data as possible from the show's platform API. Rather than doing "if platform==youtube, call youtube_search()", each platform object is assigned a standardized API class (possibly via a settings dict) with a get_initial() method.

This will help us not have to add "elif platform==newplatform" a million times when we want to add a new platform to the system.

TL;DR: All "if platform==...elif platform==" blocks should become object.platform.get_api_class().method_needed()

**Comment by gwhigs** [comment]
Fixed with commit fb1204de99ff76554d9509bd41324cb9d841ac60
